web/index.html

```diff
@@ -552,64 +601,147 @@ <h3>Used in</h3>
                     };
 
                     const animateScroll = (currentTime) => {
+                        if (this.isScrolling) {
+                            return;
+                        }
                         const timeElapsed = currentTime - startTime;
-                        const run = easeInOutQuad(timeElapsed, start, position - start, duration);
-
+                        const run = easeInOutQuad(timeElapsed, start, 0 - start, duration);
                         window.scrollTo(0, run);
-
                         if (timeElapsed < duration) requestAnimationFrame(animateScroll);
                     };
 
                     requestAnimationFrame(animateScroll);
                 },
                 clearSaveData: function () {
-                    this.saveData = [];
+                    this.saveData = new Set();
+                    this.saveDataArray = [];
                     localStorage.setItem("saveData", "");
+                    window.location.reload();
+                },
+                dragStart() {
+                    this.isDraggingBookmarklet = true;
+                },
+                dragEnd() {
+                    this.isDraggingBookmarklet = false;
+                },
+                preventClick(event) {
+                    this.isShaking = true;
+                    setTimeout(() => {
+                        this.isShaking = false;
+                    }, 250);
+                    event.preventDefault();
                 },
```

Comment:
This is used to shake the bookmarklet if the user tries just clicking on it, this was very helpful for the people who tested it, because almost all of them just tried clicking it like a button first, and it giving them feedback that it was not a button and you needed to do something other than click was very good